bentoml/cli/aws_sagemaker.py

```diff
@@ -0,0 +1,394 @@
+                    result.status
+                )
+                _echo(
+                    f'Failed to updated deployment {name}.'
+                    f'{error_code}:{error_message}'
+                )
+            track_cli('deploy-update-success', PLATFORM_NAME)
+            _echo(
+                f'Successfully updated Sagemaker deployment {name}', CLI_COLOR_SUCCESS
+            )
+            _print_deployment_info(result.deployment, output)
+        except BentoMLException as e:
+            _echo(
+                f'Failed to update Sagemaker deployment {name}: {str(e)}',
+                CLI_COLOR_ERROR,
+            )
+
+    @aws_sagemaker.command(help='Delete AWS Sagemaker deployment')
+    @click.argument('name', type=click.STRING)
+    @click.option(
+        '--namespace',
+        type=click.STRING,
+        help='Deployment namespace managed by BentoML, default value is "dev" which'
+        'can be changed in BentoML configuration file',
+    )
+    @click.option(
+        '--force',
+        is_flag=True,
+        help='force delete the deployment record in database and '
+        'ignore errors when deleting cloud resources',
+    )
+    def delete(name, namespace, force):
+        yatai_client = YataiClient()
+        get_deployment_result = yatai_client.deployment.get(namespace, name)
+        if get_deployment_result.status.status_code != status_pb2.Status.OK:
+            error_code, error_message = parse_pb_response_error_message(
+                get_deployment_result.status
+            )
+            _echo(
+                f'Failed to get Sagemaker deployment {name} for deletion. '
+                f'{error_code}:{error_message}',
+                CLI_COLOR_ERROR,
+            )
+            return
+        track_cli('deploy-delete', PLATFORM_NAME)
+        result = yatai_client.deployment.delete(name, namespace, force)
+        if result.status.status_code != status_pb2.Status.OK:
+            error_code, error_message = parse_pb_response_error_message(result.status)
+            _echo(
+                f'Failed to delete Sagemaker deployment {name}. '
+                f'{error_code}:{error_message}',
+                CLI_COLOR_ERROR,
+            )
+            return
+        extra_properties = {}
+        if get_deployment_result.deployment.created_at:
+            stopped_time = datetime.utcnow()
+            extra_properties['uptime'] = int(
+                (
+                    stopped_time
+                    - get_deployment_result.deployment.created_at.ToDatetime()
+                ).total_seconds()
+            )
+        track_cli('deploy-delete-success', PLATFORM_NAME, extra_properties)
+        _echo(f'Successfully deleted Sagemaker deployment "{name}"', CLI_COLOR_SUCCESS)
+
+    @aws_sagemaker.command(help='Get AWS Sagemaker deployment information')
+    @click.option('-n', '--name', type=click.STRING, help='Deployment name')
```

Comment:
just realized the "optional argument" thing with click applies here as well. maybe we should consider `list` instead.

```
bentoml aws_sagemaker list
bentoml aws_sagemaker get my_deployment_abc 
bentoml aws_sagemaker delete my_deployment_abc 
bentoml aws_sagemaker create my_deployment_abc -b=IrisClassifier:20191126125258_4AB1D4 --region=us-west-2
bentoml aws_sagemaker udpate my_deployment_abc -b=IrisClassifier:20191126123124_25DAE4

```

Reply:
reason different from other deployment commands(create/update/delete) is you could just call `get` and get list of sagemaker deployment.  That's why the name is option instead of argument here.